File: android/sslcommerzlibrary/src/main/java/com/sslwireless/sslcommerzlibrary/model/response/SSLCTransactionInfoModel.java
package com.sslwireless.sslcommerzlibrary.model.response;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class SSLCTransactionInfoModel {

    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("sessionkey")
    @Expose
    private String sessionkey;
    @SerializedName("tran_date")
    @Expose
    private String tranDate;
    @SerializedName("tran_id")
    @Expose
    private String tranId;
    @SerializedName("val_id")
    @Expose
    private String valId;
    @SerializedName("amount")
    @Expose
    private String amount;
    @SerializedName("store_amount")
    @Expose
    private String storeAmount;
    @SerializedName("bank_tran_id")
    @Expose
    private String bankTranId;
    @SerializedName("card_type")
    @Expose
    private String cardType;
    @SerializedName("card_no")
    @Expose
    private String cardNo;
    @SerializedName("card_issuer")
    @Expose
    private String cardIssuer;
    @SerializedName("card_brand")
    @Expose
    private String cardBrand;
    @SerializedName("card_issuer_country")
    @Expose
    private String cardIssuerCountry;
    @SerializedName("card_issuer_country_code")
    @Expose
    private String cardIssuerCountryCode;
    @SerializedName("currency_type")
    @Expose
    private String currencyType;
    @SerializedName("currency_amount")
    @Expose
    private String currencyAmount;
    @SerializedName("currency_rate")
    @Expose
    private String currencyRate;
    @SerializedName("base_fair")
    @Expose
    private String baseFair;
    @SerializedName("value_a")
    @Expose
    private String valueA;
    @SerializedName("value_b")
    @Expose
    private String valueB;
    @SerializedName("value_c")
    @Expose
    private String valueC;
    @SerializedName("value_d")
    @Expose
    private String valueD;
    @SerializedName("risk_title")
    @Expose
    private String riskTitle;
    @SerializedName("risk_level")
    @Expose
    private String riskLevel;
    @SerializedName("APIConnect")
    @Expose
    private String aPIConnect;
    @SerializedName("validated_on")
    @Expose
    private String validatedOn;
    @SerializedName("gw_version")
    @Expose
    private String gwVersion;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getSessionkey() {
        return sessionkey;
    }

    public void setSessionkey(String sessionkey) {
        this.sessionkey = sessionkey;
    }

    public String getTranDate() {
        return tranDate;
    }

    public void setTranDate(String tranDate) {
        this.tranDate = tranDate;
    }

    public String getTranId() {
        return tranId;
    }

    public void setTranId(String tranId) {
        this.tranId = tranId;
    }

    public String getValId() {
        return valId;
    }

    public void setValId(String valId) {
        this.valId = valId;
    }

    public String getAmount() {
        return amount;
    }

    public void setAmount(String amount) {
        this.amount = amount;
    }

    public String getStoreAmount() {
        return storeAmount;
    }

    public void setStoreAmount(String storeAmount) {
        this.storeAmount = storeAmount;
    }

    public String getBankTranId() {
        return bankTranId;
    }

    public void setBankTranId(String bankTranId) {
        this.bankTranId = bankTranId;
    }

    public String getCardType() {
        return cardType;
    }

    public void setCardType(String cardType) {
        this.cardType = cardType;
    }

    public String getCardNo() {
        return cardNo;
    }

    public void setCardNo(String cardNo) {
        this.cardNo = cardNo;
    }

    public String getCardIssuer() {
        return cardIssuer;
    }

    public void setCardIssuer(String cardIssuer) {
        this.cardIssuer = cardIssuer;
    }

    public String getCardBrand() {
        return cardBrand;
    }

    public void setCardBrand(String cardBrand) {
        this.cardBrand = cardBrand;
    }

    public String getCardIssuerCountry() {
        return cardIssuerCountry;
    }

    public void setCardIssuerCountry(String cardIssuerCountry) {
        this.cardIssuerCountry = cardIssuerCountry;
    }

    public String getCardIssuerCountryCode() {
        return cardIssuerCountryCode;
    }

    public void setCardIssuerCountryCode(String cardIssuerCountryCode) {
        this.cardIssuerCountryCode = cardIssuerCountryCode;
    }

    public String getCurrencyType() {
        return currencyType;
    }

    public void setCurrencyType(String currencyType) {
        this.currencyType = currencyType;
    }

    public String getCurrencyAmount() {
        return currencyAmount;
    }

    public void setCurrencyAmount(String currencyAmount) {
        this.currencyAmount = currencyAmount;
    }

    public String getCurrencyRate() {
        return currencyRate;
    }

    public void setCurrencyRate(String currencyRate) {
        this.currencyRate = currencyRate;
    }

    public String getBaseFair() {
        return baseFair;
    }

    public void setBaseFair(String baseFair) {
        this.baseFair = baseFair;
    }

    public String getValueA() {
        return valueA;
    }

    public void setValueA(String valueA) {
        this.valueA = valueA;
    }

    public String getValueB() {
        return valueB;
    }

    public void setValueB(String valueB) {
        this.valueB = valueB;
    }

    public String getValueC() {
        return valueC;
    }

    public void setValueC(String valueC) {
        this.valueC = valueC;
    }

    public String getValueD() {
        return valueD;
    }

    public void setValueD(String valueD) {
        this.valueD = valueD;
    }

    public String getRiskTitle() {
        return riskTitle;
    }

    public void setRiskTitle(String riskTitle) {
        this.riskTitle = riskTitle;
    }

    public String getRiskLevel() {
        return riskLevel;
    }

    public void setRiskLevel(String riskLevel) {
        this.riskLevel = riskLevel;
    }

    public String getAPIConnect() {
        return aPIConnect;
    }

    public void setAPIConnect(String aPIConnect) {
        this.aPIConnect = aPIConnect;
    }

    public String getValidatedOn() {
        return validatedOn;
    }

    public void setValidatedOn(String validatedOn) {
        this.validatedOn = validatedOn;
    }

    public String getGwVersion() {
        return gwVersion;
    }

    public void setGwVersion(String gwVersion) {
        this.gwVersion = gwVersion;
    }
}

File: lib/model/SSLCTransactionInfoModel.dart
class SSLCTransactionInfoModel {
  String aPIConnect;
  String amount;
  String bankTranId;
  String baseFair;
  String cardBrand;
  String cardIssuer;
  String cardIssuerCountry;
  String cardIssuerCountryCode;
  String cardNo;
  String cardType;
  String currencyAmount;
  String currencyRate;
  String currencyType;
  String gwVersion;
  String riskLevel;
  String riskTitle;
  String sessionkey;
  String status;
  String storeAmount;
  String tranDate;
  String tranId;
  String valId;
  String validatedOn;
  String valueA;
  String valueB;
  String valueC;
  String valueD;

  SSLCTransactionInfoModel(
      {this.aPIConnect,
      this.amount,
      this.bankTranId,
      this.baseFair,
      this.cardBrand,
      this.cardIssuer,
      this.cardIssuerCountry,
      this.cardIssuerCountryCode,
      this.cardNo,
      this.cardType,
      this.currencyAmount,
      this.currencyRate,
      this.currencyType,
      this.gwVersion,
      this.riskLevel,
      this.riskTitle,
      this.sessionkey,
      this.status,
      this.storeAmount,
      this.tranDate,
      this.tranId,
      this.valId,
      this.validatedOn,
      this.valueA,
      this.valueB,
      this.valueC,
      this.valueD});

  SSLCTransactionInfoModel.fromJson(Map<String, dynamic> json) {
    aPIConnect = json['APIConnect'];
    amount = json['amount'];
    bankTranId = json['bank_tran_id'];
    baseFair = json['base_fair'];
    cardBrand = json['card_brand'];
    cardIssuer = json['card_issuer'];
    cardIssuerCountry = json['card_issuer_country'];
    cardIssuerCountryCode = json['card_issuer_country_code'];
    cardNo = json['card_no'];
    cardType = json['card_type'];
    currencyAmount = json['currency_amount'];
    currencyRate = json['currency_rate'];
    currencyType = json['currency_type'];
    gwVersion = json['gw_version'];
    riskLevel = json['risk_level'];
    riskTitle = json['risk_title'];
    sessionkey = json['sessionkey'];
    status = json['status'];
    storeAmount = json['store_amount'];
    tranDate = json['tran_date'];
    tranId = json['tran_id'];
    valId = json['val_id'];
    validatedOn = json['validated_on'];
    valueA = json['value_a'];
    valueB = json['value_b'];
    valueC = json['value_c'];
    valueD = json['value_d'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['APIConnect'] = this.aPIConnect;
    data['amount'] = this.amount;
    data['bank_tran_id'] = this.bankTranId;
    data['base_fair'] = this.baseFair;
    data['card_brand'] = this.cardBrand;
    data['card_issuer'] = this.cardIssuer;
    data['card_issuer_country'] = this.cardIssuerCountry;
    data['card_issuer_country_code'] = this.cardIssuerCountryCode;
    data['card_no'] = this.cardNo;
    data['card_type'] = this.cardType;
    data['currency_amount'] = this.currencyAmount;
    data['currency_rate'] = this.currencyRate;
    data['currency_type'] = this.currencyType;
    data['gw_version'] = this.gwVersion;
    data['risk_level'] = this.riskLevel;
    data['risk_title'] = this.riskTitle;
    data['sessionkey'] = this.sessionkey;
    data['status'] = this.status;
    data['store_amount'] = this.storeAmount;
    data['tran_date'] = this.tranDate;
    data['tran_id'] = this.tranId;
    data['val_id'] = this.valId;
    data['validated_on'] = this.validatedOn;
    data['value_a'] = this.valueA;
    data['value_b'] = this.valueB;
    data['value_c'] = this.valueC;
    data['value_d'] = this.valueD;
    return data;
  }
}
